src/java/org/apache/cassandra/db/filter/IndexHints.java

```diff
@@ -0,0 +1,674 @@
+    {
+        this.included = included;
+        this.excluded = excluded;
+    }
+
+    /**
+     * @param index an index
+     * @return {@code true} if the index is included, {@code false} otherwise
+     */
+    public boolean includes(Index index)
+    {
+        return includes(index.getIndexMetadata().name);
+    }
+
+    /**
+     * @param indexName the name of an index
+     * @return {@code true} if the index is included, {@code false} otherwise
+     */
+    public boolean includes(String indexName)
+    {
+        for (IndexMetadata i : included)
+        {
+            if (i.name.equals(indexName))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns the indexes in the specified collection of indexes that are included by these hints.
+     *
+     * @param indexes a collection of indexes
+     * @return the indexes that are included by these hints
+     */
+    public Set<Index> includedIn(Collection<Index> indexes)
+    {
+        Set<Index> result = new HashSet<>();
+        for (Index index : indexes)
+        {
+            if (includes(index))
+                result.add(index);
+        }
+        return result;
+    }
+
+    /**
+     * @param indexes a collection of indexes
+     * @return {@code true} if any of the indexes is included, {@code false} otherwise
+     */
+    public boolean includesAnyOf(Collection<Index> indexes)
+    {
+        for (Index index : indexes)
+        {
+            if (includes(index))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param index an index
+     * @return {@code true} if the index is excluded, {@code false} otherwise
+     */
+    public boolean excludes(Index index)
+    {
+        return excludes(index.getIndexMetadata().name);
+    }
+
+    /**
+     * @param indexName the name of an index
+     * @return {@code true} if the index is excluded, {@code false} otherwise
+     */
+    public boolean excludes(String indexName)
+    {
+        for (IndexMetadata i : excluded)
+        {
+            if (i.name.equals(indexName))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param indexes a set of indexes
+     * @return the indexes that are not excluded by these hints
+     */
+    public <T extends Index> Set<T> notExcluded(Iterable<T> indexes)
+    {
+        Set<T> result = new HashSet<>();
+        for (T index : indexes)
+        {
+            if (!excludes(index))
+                result.add(index);
+        }
+        return result;
+    }
+
+    /**
+     * Returns the best of the specified indexes that satisfies the specified filter, according to these hints first,
+     * and the index-provided selectivity second.
+     *
+     * @param indexes a collection of indexes
+     * @param filter a filter to apply to the indexes
+     * @param isContains whether the calling operation is a {@code [NOT] CONTAINS [KEY]} operation, in which case we
```

Comment:
What is the correct terminology to use: operator or operation? I see that the file is named `Operator.java`.